### PR TITLE
Fix missing softnewline

### PR DESCRIPTION
**Pull request by nickfujita**
Resolves issue #103 
Soft newline characters (\n) were being rendered as plain text. This PR adds support for soft newline styling.

### CODE REVIEW

**Comment by Arnavion** [comment]
You also need to make the renderer understand it. As it is right now, the renderer will not emit it at all, whereas you want it to be emitted either as a space or a hard newline based on the wrapping mode.

https://github.com/Arnavion/libjass/blob/e0c630038b9738fde8077d4cef38f2f547d08006/src/renderers/web/renderer.ts#L943-L945

**Comment by nickfujita** [comment]
Added render logic to render SoftNewLine same as NewLine when WrapStyle is NoLineWrapping, and as a space for any other WrapStyle.

File: src/renderers/web/renderer.ts [hunk 1]
```diff
@@ -943,6 +943,14 @@ export class WebRenderer extends NullRenderer implements EventSource<string> {
 			else if (part instanceof parts.NewLine) {
 				startNewSpan(true);
 			}
+
+			else if (part instanceof parts.SoftNewLine && wrappingStyle === WrappingStyle.NoLineWrapping) {
+				startNewSpan(true);
+			}
+
+			else if (part instanceof parts.SoftNewLine) {
+				currentSpan!.appendChild(document.createTextNode("\u200C"));
```
Comment (Arnavion) [inline src/renderers/web/renderer.ts]: It's supposed to be a space, not a ZWNJ.

Comment (nickfujita) [reply src/renderers/web/renderer.ts]: Updated to space.

**Comment by nickfujita** [comment]
@Arnavion Wanted to followup with you to see if there is any additional feedback on this change request.

**Comment by nickfujita** [comment]
@Arnavion Sorry to ping again, but wanted to followup with you to see if there is any additional feedback on this change request before it can be merged?

**Pull request closed without merging**